rasa/nlu/tokenizers/tokenizer.py

```diff
@@ -174,7 +184,10 @@ def _split_name(self, message: Message, attribute: Text = INTENT) -> List[Token]
         else:
             words = self._tokenize_on_split_symbol(text)
 
-        return self._convert_words_to_tokens(words, text)
+        if prefix is not None:
+            words = [prefix] + words
```

Comment:
prefix should probably also be tokenized: "transfer_money!send" -> ["transfer", "money", "send"]